### PR TITLE
Tiny fixes and refactoring

### PR DESCRIPTION
**Pull request by miha2733**
- [refactor] simpleConditionalExecution -> statelessConditionalExecution
- [refactor] reduceConditionalExecution -> statedConditionalExecution
- error message was fixed
- instantiators merging was fixed

### CODE REVIEW

**Comment by dvvrd** [comment]
> Fix mono version

A vot i net

**Comment by Columpio** [comment]
`sed -iE 's|\(\t\| System.Environment.mono_corlib_version ~>\)|\t\| System.Environment.<Platform>k__BackingField ~> Win32S\n\1|g' VSharp.Test/Tests/VSharp.CSharpUtils/Unix.gold`

File: VSharp.SILI.Core/Operators.fs [hunk 1]
```diff
@@ -94,4 +87,4 @@ module internal Operators =
                     simplifyHeapPointwiseEquality mtd l1 l2;
                     eqTypes mtd t1 t2 id
                 ]
-        | _ -> internalfail "not array!"
+        | term1, term2 -> internalfail "expected array and array, but %O and %O got!" term1 term2
```
Comment (Columpio) [inline VSharp.SILI.Core/Operators.fs]: `internalfailf` maybe?

Comment (Columpio) [reply VSharp.SILI.Core/Operators.fs]: also no comma before but, it's not Russian

**Pull request closed without merging**